nilearn/plotting/displays/_figures.py

```diff
@@ -184,6 +186,9 @@ def add_contours(
             Regions are traced by connecting the centroids of non-isolated
             faces (triangles).
         """
+        if isinstance(roi_map, SurfaceImage):
+            roi_map = roi_map.data.parts[hemi]
+
```

Comment:
Is it correct ?

Reply:
yes AFAICT

the issue is in case people pass for the contours an image whose mesh does not match the on they passed when they first plotted the stat map.

```python
figure = plot_surf_stat_map(
    stat_map=img,
    surf_mesh=fsaverage_meshes["inflated"],
    hemi="right",
    engine="plotly",
)

figure.add_contours(
        roi_map=surface_image_with_incompatible_mesh,
        hemi="right",
    )

# or wrong hemisphere
figure.add_contours(
        roi_map=this_surface_image_is_ok,
        hemi="left", # wrong hemisphere
    )
```

Not sure we can properly do some error handling in this case.